shipyard_blueprints/bigquery/shipyard_bigquery/shipyard/upload.py

```diff
@@ -0,0 +1,131 @@
+    parser.add_argument(
+        "--upload-type",
+        dest="upload_type",
+        default="append",
+        choices={"append", "overwrite"},
+        required=False,
+    )
+    parser.add_argument(
+        "--source-file-name-match-type",
+        dest="source_file_name_match_type",
+        default="exact_match",
+        choices={"exact_match", "regex_match"},
+        required=False,
+    )
+    parser.add_argument("--source-file-name", dest="source_file_name", required=True)
+    parser.add_argument(
+        "--source-folder-name", dest="source_folder_name", default="", required=False
+    )
+    parser.add_argument("--schema", dest="schema", default="", required=False)
+    parser.add_argument(
+        "--skip-header-rows", dest="skip_header_rows", default="", required=False
+    )
+    parser.add_argument(
+        "--quoted-newline", dest="quoted_newline", default="FALSE", required=False
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    try:
+        args = get_args()
+        dataset = args.dataset
+        table = args.table
+        upload_type = args.upload_type
+        file_name = args.source_file_name
+        folder_name = args.source_folder_name
+        full_path = shipyard.files.combine_folder_and_file_name(
+            folder_name=f"{os.getcwd()}/{folder_name}", file_name=file_name
+        )
+        match_type = args.source_file_name_match_type
+        schema = None if args.schema == "" else ast.literal_eval(args.schema)
+        quoted_newline = shipyard.args.convert_to_boolean(args.quoted_newline)
+
+        skip_header_rows = (
+            None if args.skip_header_rows == "" else args.skip_header_rows
+        )
+
+        if skip_header_rows:
+            skip_header_rows = int(args.skip_header_rows)
+
+        if schema and not skip_header_rows:
+            logger.warning(
+                f"Skip Header Rows was not provided but a schema was defined. Setting the Skip Header Rows to 1 to properly load"
+            )
+            skip_header_rows = 1
+
+        client = BigQueryClient(args.service_account)
+        client.connect()
+        logger.info("Successfully connected to BigQuery")
+
+        if match_type == "regex_match":
+            file_names = shipyard.files.find_all_local_file_names(folder_name)
+            matching_file_names = shipyard.files.find_all_file_matches(
+                file_names, re.compile(file_name)
+            )
+            logger.info(
+                f"{len(matching_file_names)} files found. Preparing to upload..."
+            )
+
+            for index, file_name in enumerate(matching_file_names, start=1):
+                logger.info(f"Uploading file {index} of {len(matching_file_names)}")
+                client.upload(
+                    file=file_name,
+                    dataset=dataset,
+                    table=table,
+                    upload_type=upload_type,
+                    skip_header_rows=skip_header_rows,
```

Comment:
Does this work as expected if `skip_header_rows=""` if so fine as is otherwise i recommend
`if args.skip_header_rows:
     skip_header_rows=int(args.skip_header_rows)
  else:
  logger.warning("skip header row was not defined. Continuing with default value of 1")
  ```

Reply:
This does work because it is set to `None` above, and the method defined in the class defaults to None